src/litserve/server.py

```diff
@@ -422,6 +424,8 @@ def __init__(
         stream: bool = False,
         spec: Optional[LitSpec] = None,
         max_payload_size=None,
+        middlewares: Optional[list[tuple[type[_MiddlewareClass[P]]], dict]] = None,
```

Comment:
`|` operator is not supported in Python 3.8 I think so you might need to use `typing.Union`.